### PR TITLE
Bump traitlets to version 5

### PR DESCRIPTION
**Pull request by maxnoe**

### CODE REVIEW

**Comment by maxnoe** [comment]
Traitlets 5 requires python 3.7+, so we need to merge #1467

File: ctapipe/tools/tests/test_tools.py [hunk 1]
```diff
@@ -319,6 +319,8 @@ def test_plot_charge_resolution(tmpdir):
     output_path = os.path.join(str(tmpdir), "cr.pdf")
     tool = ChargeResolutionViewer()
 
-    assert run_tool(tool, ["-f", [path], "-o", output_path]) == 0
+    argv = ["-f", str(path), "-o", output_path]
+    print(argv)
```
Comment (watsonjj) [inline ctapipe/tools/tests/test_tools.py:323]: Stray print statement?

Comment (maxnoe) [reply ctapipe/tools/tests/test_tools.py:323]: yep

Comment (maxnoe) [reply ctapipe/tools/tests/test_tools.py]: fixed

File: setup.py [hunk 2]
```diff
@@ -69,7 +69,7 @@
         "scipy~=1.2",
         "tables~=3.4",
         "tqdm>=4.32",
-        "traitlets>=4.1,<5.0",
+        "traitlets~=5.0,>=5.0.5",
```
Comment (watsonjj) [inline setup.py:72]: Is this not identical to `"traitlets~=5.0.5"`?

Comment (maxnoe) [reply setup.py:72]: No, since `~=5.0.5` would fix us to `5.0.x`

**Review by watsonjj — APPROVED** [review]

**Review by kosack — APPROVED** [review]

**Pull request merged**